File: src/extension/Extensions.java
package extension;

import java.util.List;

public class Extensions {

    public static void main(String[] args) {
        final var c = ExtensionsKt.lastChar("Happy Jordi");
        System.out.println(c);

        System.out.println(String.join(",", "a", "b", "c"));

        final var array = "12.345-6.A".split("\\.");
        for (int i = 0; i < array.length; i++) {
            System.out.print(array[i] + "\t");
        }
    }
}

File: src/extension/Extensions.kt
package extension

fun main() {
    val c = "Happy Bear".lastChar();
    println(c)

    val cat = Cat("ChunBae")
    val lastChatOfName = cat.getName().lastChar()
    println(lastChatOfName)

    println("abc".getStar(1))

    val cat2: Animal = Cat("Nabi")
    cat2.talk()
}

// String -> 작성된 확장 함수의 Receiver
fun String.lastChar() = get(length - 1)

fun String.getStar(index: Int) = "*"

fun Animal.talk() {
    println("myang")
}

fun Cat.talk() {
    println("meow")
}